rueidiscompat/command.go

```diff
@@ -2109,6 +2112,134 @@ func (cmd *ClusterSlotsCmd) Result() ([]ClusterSlot, error) {
 	return cmd.val, cmd.err
 }
 
+func newClusterShardsCmd(res rueidis.RedisResult) *ClusterShardsCmd {
+	arr, err := res.ToArray()
+	if err != nil {
+		return &ClusterShardsCmd{err: err}
+	}
+	val := make([]ClusterShard, 0, len(arr))
+	for _, v := range arr {
+		dict, err := v.ToMap()
+		if err != nil {
+			return &ClusterShardsCmd{err: err}
+		}
+		var (
+			start, end int64
+			nodesArr   []ClusterNode
+		)
+		{
+			slots, ok := dict["slots"]
+			if !ok {
+				return &ClusterShardsCmd{err: errors.New("slots not found")}
+			}
+			arr, err := slots.ToArray()
+			if err != nil {
+				return &ClusterShardsCmd{err: err}
+			}
+			if len(arr) != 2 {
```

Comment:
We can relax the condition to be
```suggestion
			if len(arr) >= 2 {
						start = arr[0]
						end = arr[1]
```

Reply:
goredis seems to keep track of not just first two slots. I aligned it with that now.